contracts/token/ERC7984/extensions/ERC7984Rwa.sol

```diff
@@ -0,0 +1,245 @@
+        _setConfidentialFrozen(account, FHE.fromExternal(encryptedAmount, inputProof));
+    }
+
+    /// @dev Mints confidential amount of tokens to account with proof.
+    function confidentialMint(
+        address to,
+        externalEuint64 encryptedAmount,
+        bytes calldata inputProof
+    ) public virtual onlyAgent returns (euint64) {
+        return _confidentialMint(to, FHE.fromExternal(encryptedAmount, inputProof));
+    }
+
+    /// @dev Mints confidential amount of tokens to account.
+    function confidentialMint(address to, euint64 encryptedAmount) public virtual onlyAgent returns (euint64) {
+        return _confidentialMint(to, encryptedAmount);
+    }
+
+    /// @dev Burns confidential amount of tokens from account with proof.
+    function confidentialBurn(
+        address account,
+        externalEuint64 encryptedAmount,
+        bytes calldata inputProof
+    ) public virtual onlyAgent returns (euint64) {
+        return _confidentialBurn(account, FHE.fromExternal(encryptedAmount, inputProof));
+    }
+
+    /// @dev Burns confidential amount of tokens from account.
+    function confidentialBurn(address account, euint64 encryptedAmount) public virtual onlyAgent returns (euint64) {
+        return _confidentialBurn(account, encryptedAmount);
+    }
+
+    /// @dev Forces transfer of confidential amount of tokens from account to account with proof by skipping compliance checks.
+    function forceConfidentialTransferFrom(
+        address from,
+        address to,
+        externalEuint64 encryptedAmount,
+        bytes calldata inputProof
+    ) public virtual onlyAgent returns (euint64) {
+        return _forceConfidentialTransferFrom(from, to, FHE.fromExternal(encryptedAmount, inputProof));
+    }
+
+    /// @dev Forces transfer of confidential amount of tokens from account to account by skipping compliance checks.
+    function forceConfidentialTransferFrom(
+        address from,
+        address to,
+        euint64 encryptedAmount
+    ) public virtual onlyAgent returns (euint64 transferred) {
+        return _forceConfidentialTransferFrom(from, to, encryptedAmount);
+    }
+
+    /// @dev Internal function which mints confidential amount of tokens to account.
+    function _confidentialMint(address to, euint64 encryptedAmount) internal virtual returns (euint64) {
+        return _mint(to, encryptedAmount);
+    }
+
+    /// @dev Internal function which burns confidential amount of tokens from account.
+    function _confidentialBurn(address account, euint64 encryptedAmount) internal virtual returns (euint64) {
+        return _burn(account, encryptedAmount);
+    }
+
+    /// @dev Internal function which forces transfer of confidential amount of tokens from account to account by skipping compliance checks.
+    function _forceConfidentialTransferFrom(
+        address from,
+        address to,
+        euint64 encryptedAmount
+    ) internal virtual returns (euint64 transferred) {
```

Comment:
should this be private?

Reply:
if we keep the hooks we can make this private. if the hooks are removed this should be internal.